src/main/java/WayofTime/alchemicalWizardry/common/items/SacrificialDagger.java

```diff
@@ -58,93 +63,102 @@ public void addInformation(ItemStack stack, EntityPlayer par2EntityPlayer, List
      */
     @Override
     public void onPlayerStoppedUsing(ItemStack stack, World world, EntityPlayer player, int itemInUseCount) {
-        // if(itemInUseCount < 32)
-        // {
-        // return;
-        // }
+        int usageCount = this.getMaxItemUseDuration(stack) - itemInUseCount;
+        int hpCount = (int) Math.min(((usageCount * 0.35)), player.getMaxHealth());
 
-        PlayerSacrificeHandler.sacrificePlayerHealth(player);
-    }
+        Random rnd = new Random();
+        int shuffle = rnd.nextInt(3);
 
-    @Override
-    public int getMaxItemUseDuration(ItemStack stack) {
-        return 72000;
-    }
+        if (player instanceof FakePlayer) {
+            return;
+        }
 
-    /**
-     * returns the action that specifies what animation to play when the items is being used
-     */
-    @Override
-    public EnumAction getItemUseAction(ItemStack stack) {
-        return EnumAction.bow;
-    }
+        if (player.isPotionActive(AlchemicalWizardry.customPotionSoulFray)) {
+            if (world.isRemote) player.addChatComponentMessage(
+                    new ChatComponentTranslation("message.sacrifice.soulfray" + shuffle)
+                            .setChatStyle(new ChatStyle().setColor(EnumChatFormatting.RED)));
+            return;
+        }
 
-    @Override
-    public ItemStack onItemRightClick(ItemStack stack, World world, EntityPlayer player) {
-        if (this.canUseForSacrifice(stack)) {
-            player.setItemInUse(stack, this.getMaxItemUseDuration(stack));
-            return stack;
+        if (hpCount < 1) {
```

Comment:
I hate this, players should be allowed to hurt themselves even if it would kill themselves